### PR TITLE
Parking api

### PR DESCRIPTION
**Pull request by aangelisc**
- Introducing an API to handle car parking spaces

### CODE REVIEW

File: server/bookings/createBooking.ts [hunk 1]
```diff
@@ -72,7 +72,8 @@ export const createBooking = async (
   const officeBookedSuccessfully = await incrementOfficeBookingCount(
     config,
     requestedOffice,
-    request.date
+    request.date,
+    request.includesParking
```
Comment (aangelisc) [inline server/bookings/createBooking.ts:76]: Change naming to parking (keep it short)

File: server/db/officeBookings.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ export interface OfficeBooking {
   name: string;
   date: string;
   bookingCount: number;
+  parkingBookingCount: number;
```
Comment (aangelisc) [inline server/db/officeBookings.ts:20]: Change to parkingCount

File: server/db/officeBookings.ts [hunk 3]
```diff
@@ -42,7 +45,8 @@ const buildMapper = (config: Config) =>
 export const incrementOfficeBookingCount = async (
   config: Config,
   office: OfficeQuota,
-  date: string
+  date: string,
+  includesParking: boolean
```
Comment (aangelisc) [inline server/db/officeBookings.ts:49]: includeParking (present tense)

File: server/db/officeBookings.ts [hunk 4]
```diff
@@ -76,17 +81,30 @@ export const incrementOfficeBookingCount = async (
     new MathematicalExpression(new AttributePath('bookingCount'), '+', 1)
   );
 
+  let parkingQuotaValue;
+
+  if (includesParking) {
```
Comment (aangelisc) [inline server/db/officeBookings.ts:86]: includeParking

File: server/db/officeBookings.ts [hunk 5]
```diff
@@ -111,6 +129,10 @@ export const decrementOfficeBookingCount = async (
     'bookingCount',
     new MathematicalExpression(new AttributePath('bookingCount'), '-', 1)
   );
+  updateExpression.set(
```
Comment (aangelisc) [inline server/db/officeBookings.ts:132]: Need to check if original booking includes parking as well

File: test.sh [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (aangelisc) [inline test.sh:1]: Fill out test script to encompass typescript testing, client build, docker set up etc

Comment (aangelisc) [reply test.sh:1]: Remove docker-compose down in test script and prefix test tables and add supporting function to remove test tables

File: server/app-config.ts [hunk 7]
```diff
@@ -23,7 +23,8 @@ const isOfficeQuotas = (input: any): input is OfficeQuota[] =>
       typeof o === 'object' &&
       o !== null &&
       typeof o.name === 'string' &&
-      typeof o.quota === 'number'
+      typeof o.quota === 'number' &&
+      typeof o.parkingQuota === 'number'
```
Comment (danielrbradley) [inline server/app-config.ts:27]: We should default this to zero if not provided in the ENV (after merging/rebasing with changes on master)

File: server/bookings/model.ts [hunk 8]
```diff
@@ -5,10 +5,14 @@ export type CreateBooking = {
...
   office: string;
+  parking: boolean;
 };
 
 export const isCreateBooking = (arg: any): arg is CreateBooking =>
-  typeof arg.user === 'string' && typeof arg.date === 'string' && typeof arg.office === 'string';
+  typeof arg.user === 'string' &&
+  typeof arg.date === 'string' &&
+  typeof arg.office === 'string' &&
+  typeof arg.parking === 'boolean';
```
Comment (danielrbradley) [inline server/bookings/model.ts:15]: On second thoughts ... adding this breaks the current client implementation - we should make this optional, at least for now and default it to false.

File: server/db/officeBookings.ts [hunk 9]
```diff
@@ -27,6 +28,8 @@ export class OfficeBookingModel {
   date!: string;
   @attribute()
   bookingCount!: number;
+  @attribute()
+  parkingCount!: number;
```
Comment (danielrbradley) [inline server/db/officeBookings.ts:32]: What happens for rows already in the DB without this attribute?

File: server/tests/test-utils.ts [hunk 10]
```diff
@@ -46,12 +49,10 @@ export const server = () => {
...
 export const resetDb = async () => {
+  const config = getConfig();
   await createLocalTables(
-    { deleteTablesFirst: true },
-    {
-      region: 'eu-west-1',
-      endpoint: 'http://localhost:8000',
-    }
+    { deleteTablesFirst: true, tableNamePrefix: config.dynamoDBTablePrefix },
+    config.dynamoDB
```
Comment (danielrbradley) [inline server/tests/test-utils.ts:55]: A thought here - should we actually make the setup (and therefore reset) isolated per test file so we can disable run-in-band?

Comment (aangelisc) [reply server/tests/test-utils.ts:57]: Will follow up in another PR

**Comment by aangelisc** [comment]
**BREAKING CHANGE**

This PR adds functionality to include booking parking spaces.

If there are any existing office bookings rows in the DB, we won't yet be able to increment or decrement the parking count on them as it will be undefined.

**Pull request merged**